src/utils/featheryClient/index.ts

```diff
@@ -72,7 +74,50 @@ export const updateRegionApiUrls = (region: string) => {
   }
 };
 
+/**
+ * The number of milliseconds waited until another submitCustom call
+ */
+const SUBMIT_CUSTOM_DEBOUNCE_WINDOW = 1000;
+
 export default class FeatheryClient extends IntegrationClient {
+  /**
+   * Used to aggregate field value updates for sucessive calls to
+   * submitCustom within the debounce window
+   */
+  pendingSubmitCustomUpdates: { [key: string]: any };
+
+  /**
+   * Debounced implementation of submitCustom
+   */
+  debouncedSubmitCustom: DebouncedFunc<(override: boolean) => Promise<void>>;
+
+  /**
+   * If there is a pending invocation of submitCustom, this method calls it immediately
+   */
+  flushPendingSubmitCustomUpdates: (
+    override?: boolean
+  ) => Promise<void> | undefined;
+
+  constructor(
+    formKey = '',
+    ignoreNetworkErrors?: any,
+    draft = false,
+    bypassCDN = false
+  ) {
+    super(formKey, ignoreNetworkErrors, draft, bypassCDN);
+    this.pendingSubmitCustomUpdates = {};
+    this.debouncedSubmitCustom = debounce(
+      this._debouncedSubmitCustom.bind(this),
+      SUBMIT_CUSTOM_DEBOUNCE_WINDOW
+    );
+    this.flushPendingSubmitCustomUpdates = (override?: boolean) => {
+      // we call the debounced method and then the flush to immediately submit changes
+      // see: https://github.com/lodash/lodash/issues/4185#issuecomment-462388355
+      this.debouncedSubmitCustom(override ?? true);
+      return this.debouncedSubmitCustom.flush();
```

Comment:
i'm not familiar with the flush function. will this return the promise from the network request so it can be awaited by the client of this function?